### DISCUSSION TITLE
Quests Requirement to rankups

### DISCUSSION THREAD

**Discussion by DrakePork** [Zrips/CMI#3456] [Feature Requests]
**Description of feature/improvement request:**
Add support for the quests plugin by PikaMug, and add the ability to have quests as a rankup requirement.

---
### Detailed explanation of what it should do, when and in what situation: ###
When a player is trying to rank up, one of the requirements for the rankup is doing a quest. So the player has to do the quest before being able to rankup.

### Basic explanation as to why this feature is needed and/or useful to have: ###
Would be useful to have if you wanted players to have to do a quest to rankup.

**Comment by CloudeLecaw** [comment]
You can setup requirements for rank up. For the quest thing i suggest a pluglin like [quests](https://www.spigotmc.org/resources/quests.3711/?__cf_chl_jschl_tk__=a4de458efb7959faf85842f048d0a2e8b094fce5-1591328965-0-AZ0f3y1XW2G3ui9vgf7z8aRzaYGUsYQUhdf0NGhTmnfTV7gZQwhnrJw0pqOGzjPrsA7Ym0ZUaN4cT39Kz9_bqeH1ptbfb9MHt7QGDA1pvL_NQTBDtpOXVnBkd8-RxK5oigi4bHhvBKcXkxbEuFtbwWQkTP2Au2dxNGfzYPb1cRdg_h5fATk4AOoNjuT3cI0mQTqBEWZhJJCnpb_SbulhB94RlBnWP5BUSB6J7lX8QUZYRubnk29QBKU7C2j2x75nX61PsaDsZkwB62s2DemOPXxBhXkjYw5Jbpua9B-tHegbn2b9kpPVf8x7betsaNp_lw) 


There u can setup quests and use the rankup command as reward

The possible cmi requirements you can find here. https://www.zrips.net/cmi/rankup-system/

**Comment by DrakePork** [comment]
This is about having the ability to have quests as a requirement for a rank up @CloudeLecaw, not adding a rankup command as a reward to the end of a quest. It also clearly states in the feckin message if you had bothered reading that "Add support for the quests plugin by PikaMug".

**Comment by CloudeLecaw** [comment]
ah so you want to design Quests in the [quest plugin](https://www.spigotmc.org/resources/quests.3711/?__cf_chl_jschl_tk__=101b99a8c406aa48802d4cf711e9b2611c2301d0-1591346564-0-Af7kt3ihRe-NSK86qKuEGGsiixDSDH75um9dGXtsAV2nxhzWRCWR3RSKaQR9pfs2rNFmwoPnqytg9A08Qf3iR93CDQx5-N77Qh4Wf6OVNCprl2-vhQwY2f4Rhddix7SF9SoQGV_c_FyQY5-3b5nfsm3uHkpOsZvAKKL6j-YnEa0FJbgfhMGc2UHTux0StwDiIVY82R2CULVjCVbvsQ7ApFWvfHR1jHCg37bGh5RZLY15hrfI-KoDB1fWt6fjgVRZur81SQp5vXCOu2l5uHqwpnFSX-RVmTkYbVhXFj_t6J3MlyybWOeKZQUoMX-nJ4SvCQ) and the rank up requirment should ask if the quest is done right?

Does  "[quests plugin by PikaMug](https://www.spigotmc.org/resources/quests.3711/?__cf_chl_jschl_tk__=101b99a8c406aa48802d4cf711e9b2611c2301d0-1591346564-0-Af7kt3ihRe-NSK86qKuEGGsiixDSDH75um9dGXtsAV2nxhzWRCWR3RSKaQR9pfs2rNFmwoPnqytg9A08Qf3iR93CDQx5-N77Qh4Wf6OVNCprl2-vhQwY2f4Rhddix7SF9SoQGV_c_FyQY5-3b5nfsm3uHkpOsZvAKKL6j-YnEa0FJbgfhMGc2UHTux0StwDiIVY82R2CULVjCVbvsQ7ApFWvfHR1jHCg37bGh5RZLY15hrfI-KoDB1fWt6fjgVRZur81SQp5vXCOu2l5uHqwpnFSX-RVmTkYbVhXFj_t6J3MlyybWOeKZQUoMX-nJ4SvCQ)" has an option to list or placeholder or a database or something where you can see that a quests is done? - i see it has a statistics commands, but do you see jsut the current quests statics there or a list of all finished and unfished quests?- i never used quests so u dont know about..

Something like that would be needed because it would not so good if CMI needed create a whole database for finished quests for each player...

**Comment by mrfloris** [comment]
If the plugin makes it possible to give the user a permission when they do not have a question, or are doing a question or have completed a question.. you could use permissionrequirement in cmi rankup to check against. Might be a cheaty way around for now.

**Comment by DrakePork** [comment]
@CloudeLecaw all quests a player completes gets saved in a file yes.

@mrfloris I did think about that yeah but would look a bit bad and be a bit confusing if it said you needed to get a permission not a quest

**Comment by CloudeLecaw** [comment]
in a file?
this is not so good, as long u cant check this with an output for each single quest if its done or not
you need a output for a hook
like quest:questname:done   yes-no
better would be a permisison but i see the quests does not support this

May you can ask the Quests dev to create a command where you can get true/false outputs for asking for a quest

**Comment by DrakePork** [comment]
It does have placeholders for if a quest is completed.
